sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusMessage.cs

```diff
@@ -108,6 +108,26 @@ public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
         /// </remarks>
         public BinaryData Body { get; set; }
 
+        internal enum AmqpBodyType
+        {
+            Unspecified,
+            Data,
+            Sequence,
+            Value
+        }
+
+        internal AmqpBodyType AmqpBodyTypeHint { get; set; }
+
+        internal ReadOnlyMemory<byte> AmqpData
+        {
+            get => Body.AsBytes();
+            set => Body = new BinaryData(value);
+        }
```

Comment:
Allows 1-1 mapping between public (existing) Body property and this AmqpData property. This was 'mapped' this way during the conversion process anyway.